docs/docs/guides/web3_migration_guide/providers_migration_guide.md

```diff
@@ -105,11 +105,12 @@ interface ReconnectOptions {
 }
 ```
 
-In 4.x, the options object is of type `ClientRequestArgs` or of `ClientOptions`. See [here](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules__types_node_http_d_._http_.clientrequestargs.html) for `ClientRequestArgs` and [here](https://github.com/websockets/ws) for `ClientOptions`.
+In 4.x, the `socketOptions` parameter is of type `ClientRequestArgs` or of `ClientOptions`. See [here](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules__types_node_http_d_._http_.clientrequestargs.html) for `ClientRequestArgs` and [here](https://github.com/websockets/ws) for `ClientOptions`.
```

Comment:
Why ClientRequestArgs is option to pass in websocket connection? That is for http module https://microsoft.github.io/PowerBI-JavaScript/modules/_node_modules__types_node_http_d_._http_.html

we are using[ isomeric-ws](https://github.com/heineiuo/isomorphic-ws) with [options](https://github.com/websockets/ws/blob/master/doc/ws.md#new-websocketaddress-protocols-options) passing support . This should be changed in code as well.

Reply:
Actually, [isomeric-ws](https://github.com/heineiuo/isomorphic-ws) uses this `ClientRequestArgs`. It is a bit strange to me also :smile: 

```ts
import {
...
    ClientRequestArgs,
...
} from "http";
...

// WebSocket socket.
declare class WebSocket extends EventEmitter {
...
    constructor(address: string | URL, options?: WebSocket.ClientOptions | ClientRequestArgs);
    constructor(
        address: string | URL,
        protocols?: string | string[],
        options?: WebSocket.ClientOptions | ClientRequestArgs,
    );
...
```
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/85754720ec0e51d48dc28a3e7230a327899e79af/types/ws/index.d.ts#L94

Reply:
Interesting